### PR TITLE
zerolog: Hook implementation

### PR DESCRIPTION
**Pull request by AkhigbeEromo**
part of #5405 
Implemented the run method and wrote tests for it

### CODE REVIEW

File: bridges/otelzerolog/hook_test.go [hunk 1]
```diff
@@ -3,13 +3,18 @@
 package otelzerolog
 
 import (
+	"os"
+	// "reflect".
```
Comment (dmathieu) [inline bridges/otelzerolog/hook_test.go:7]: Why keep those as comments? (same for `time`). If we're not using those, we should remove them.

File: bridges/otelzerolog/hook_test.go [hunk 2]
```diff
@@ -103,3 +108,81 @@ func TestNewHook(t *testing.T) {
...
+
+	logger := zerolog.New(os.Stderr).Hook(hook)
+
+	t.Run("Run", func(t *testing.T) {
+		// Create an event and run the hook
+		event := logger.Info().Str(testKey, testValue)
+		hook.Run(event, testEntry, testMessage)
+
+		// Check the results
+		got := rec.Result()[0].Records[0]
```
Comment (dmathieu) [inline bridges/otelzerolog/hook_test.go:132]: If for some reason, there are no results, you'll get a non-useful crash here.

```suggestion
		require.Len(t, rec.Result(), 1)
		require.Len(t, rec.Result()[0].Records, 1)
		got := rec.Result()[0].Records[0]
```

File: bridges/otelzerolog/hook.go [hunk 3]
```diff
@@ -100,5 +102,29 @@ func NewHook(name string, options ...Option) *Hook {
 
 // Run handles the passed record, and sends it to OpenTelemetry.
 func (h Hook) Run(e *zerolog.Event, level zerolog.Level, msg string) {
-	// TODO
+	r := log.Record{}
+	r.SetSeverity(convertLevel(level))
+	r.SetBody(log.StringValue(msg))
+	r.SetTimestamp(time.Now())
+	r.SetSeverityText(level.String())
```
Comment (dmathieu) [inline bridges/otelzerolog/hook.go:109]:
```suggestion
	r.SetSeverityText(level.String())
	
	// TODO: add support for attributes
	// This is limited by zerolog's unability to retrieve fields.
	// https://github.com/rs/zerolog/issues/493

```

**Review by dmathieu — COMMENTED** [review]
This looks nice :)

File: bridges/otelzerolog/hook_test.go (hunk 1 above)
Comment (AkhigbeEromo) [reply bridges/otelzerolog/hook_test.go:7]: oh, that was an error, was trying out some things 
Will take it out now

**Review by dmathieu — APPROVED** [review]

File: bridges/otelzerolog/hook.go [hunk 4]
```diff
@@ -100,5 +102,34 @@ func NewHook(name string, options ...Option) *Hook {
 
 // Run handles the passed record, and sends it to OpenTelemetry.
 func (h Hook) Run(e *zerolog.Event, level zerolog.Level, msg string) {
-	// TODO
+	r := log.Record{}
+	r.SetSeverity(convertLevel(level))
+	r.SetBody(log.StringValue(msg))
+	r.SetTimestamp(time.Now())
```
Comment (pellared) [inline bridges/otelzerolog/hook.go:108]:
```suggestion
```

This is not a value coming from zerolog. You can create an issue similar to https://github.com/rs/zerolog/issues/493.

The API implementations can set timestamps if they are unset.
For instance the OTel Go SDK set `ObservedTimestamp` to `time.Now()`.
See: https://github.com/open-telemetry/opentelemetry-go/blob/5fef3eb220acc4691497da329850db5daf0c2a3f/sdk/log/logger.go#L75-L78

**Review by pellared — APPROVED** [review]

File: bridges/otelzerolog/hook.go [hunk 5]
```diff
@@ -100,5 +100,33 @@ func NewHook(name string, options ...Option) *Hook {
...
 // Run handles the passed record, and sends it to OpenTelemetry.
 func (h Hook) Run(e *zerolog.Event, level zerolog.Level, msg string) {
-	// TODO
+	r := log.Record{}
+	r.SetSeverity(convertLevel(level))
+	r.SetBody(log.StringValue(msg))
+	r.SetSeverityText(level.String())
+
+	// TODO: add support for attributes
+	// This is limited by zerolog's unability to retrieve fields.
```
Comment (MrAlias) [inline bridges/otelzerolog/hook.go:109]:
```suggestion
	// This is limited by zerolog's inability to retrieve fields.
```

**Review by MrAlias — APPROVED** [review]

**Pull request merged**